lib/Sema/CSBindings.cpp

```diff
@@ -981,6 +981,10 @@ bool TypeVariableBinding::attempt(ConstraintSystem &cs) const {
     type = cs.openUnboundGenericType(UGT, locator);
     type = type->reconstituteSugar(/*recursive=*/false);
   }
+  
+  // Recording it as the current type variable binding
+  // being attempted is the system.
+  cs.recordCurrentTypeVariableBinding(TypeVar, this);
```

Comment:
A better place for this would be in `assignFixedType` because it would make it possible to track not only types attempted by the solver but also the ones which got assigned as a result of a `Bind` constraint.

Reply:
Humm... that makes sense 👍

Reply:
@xedin :))
I've come to attempt this approach(not sure if correctly 😆) but had some problems getting the binding source type locator(most times getting the typevar locator, not the source) in the places and also use the bind source locator in some places introduce some regressions in diagnostics that rely on locator information. 
This was in eab485b76652e236deac67d549f1e0aa5678dc7a if you want to check :)

Reply:
Thank you, @LucianoPAlmeida! I'll try to take a look this week.

Reply:
>  I'll try to take a look this week.

Thanks @xedin, no problem and no need to rush too :) 
I left some detailed notes in the last comment to clarify some points.